src/app/public/modules/ag-grid/ag-grid-data-manager-adapter.directive.ts

```diff
@@ -0,0 +1,207 @@
+
+  @Input()
+  private viewId: string;
+
+  @ContentChildren(AgGridAngular, { descendants: true })
+  public agGridList: QueryList<AgGridAngular>;
+
+  private currentAgGrid: AgGridAngular;
+  private dataStateSub: Subscription;
+  private viewConfig: SkyDataViewConfig;
+  private _ngUnsubscribe = new Subject();
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerSvc: SkyDataManagerService) { }
+
+  public ngAfterContentInit(): void {
+    this.viewConfig = this.dataManagerSvc.getViewById(this.viewId);
+    this.checkForAgGrid();
+
+    this.agGridList.changes
+    .pipe(takeUntil(this._ngUnsubscribe))
+    .subscribe(() => this.checkForAgGrid());
+  }
+
+  public ngOnDestroy() {
+    this._ngUnsubscribe.next();
+    this._ngUnsubscribe.complete();
+  }
+
+  private checkForAgGrid(): void {
+    const agGridCount = this.agGridList.length;
+
+    /* istanbul ignore else */
+    if (agGridCount > 1) {
+      console.error(
+        'A data view may only have one active ag-Grid child component instance at a time.'
+      );
+    } else if (agGridCount === 0) {
+      this.unregisterAgGrid();
+    } else if (this.agGridList.first !== this.currentAgGrid) {
+      this.registerAgGrid();
+    }
+  }
+
+  private unregisterAgGrid(): void {
+    this.currentAgGrid = undefined;
+
+    /* istanbul ignore if */
+    if (this.dataStateSub) {
+      this.dataStateSub.unsubscribe();
+    }
+  }
+
+  private registerAgGrid(): void {
+    this.unregisterAgGrid();
+
+    const agGrid = this.agGridList.first;
+
+    this.currentAgGrid = agGrid;
+
+    agGrid.gridReady
+    .pipe(takeUntil(this._ngUnsubscribe))
+    .subscribe(() => {
+      this.viewConfig.onSelectAllClick = this.selectAll.bind(this);
+      this.viewConfig.onClearAllClick = this.clearAll.bind(this);
+
+      this.dataManagerSvc.updateViewConfig(this.viewConfig);
+
+      this.displayColumns(this.dataManagerSvc.getCurrentDataState());
+
+      this.dataStateSub = this.dataManagerSvc.getDataStateUpdates(this.viewConfig.id)
+      .pipe(takeUntil(this._ngUnsubscribe))
+      .subscribe((dataState: SkyDataManagerState) => {
+        this.displayColumns(dataState);
+      });
+
+      agGrid.api.sizeColumnsToFit();
+    });
+
+    agGrid.columnMoved
+    .pipe(takeUntil(this._ngUnsubscribe))
+    .subscribe((event: ColumnMovedEvent) => {
+      let columnOrder = agGrid.columnApi.getAllDisplayedVirtualColumns().map(
+        col => col.getColDef().colId
+      );
+
+      if (event.source !== 'api') {
+        const dataState = this.dataManagerSvc.getCurrentDataState();
+
+        const viewState = dataState.getViewStateById(this.viewConfig.id);
+        viewState.displayedColumnIds = columnOrder;
+
+        this.dataManagerSvc.updateDataState(
+          dataState.addOrUpdateView(this.viewConfig.id, viewState),
+          this.viewConfig.id
+        );
+      }
+    });
+
+    agGrid.rowSelected
+    .pipe(takeUntil(this._ngUnsubscribe))
```

Comment:
When we drop to the next line and invoke a method or reference a property, we usually indent that next line.

```
    agGrid.rowSelected
      .pipe(takeUntil(this._ngUnsubscribe))
```